docs/static/config-details.asciidoc

```diff
@@ -82,10 +85,53 @@ info, see <<profiling-the-heap>>.
 ==== Setting the off-heap size
 
 The operating system, persistent queue mmap pages, direct memory, and other processes require memory in addition to memory allocated to heap size.
-Keep the overall memory requirements in mind when you allocate memory.
 
+Internal JVM data structures, thread stacks, memory mapped files, direct memory for input/output (IO) operations, and the Java heap used to execute Logstash all use JVM memory.
```

Comment:
After tech review is complete, we can work together to refine final wording. You've got some good info here, and it deserves all the clarity we can give it. :-)